tracer/src/Datadog.Trace/IAST/Analyzers/HardcodedSecretsAnalyzer.cs

```diff
@@ -0,0 +1,227 @@
+// <copyright file="HardcodedSecretsAnalyzer.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Datadog.Trace.ClrProfiler;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.Iast.Analyzers;
+
+internal class HardcodedSecretsAnalyzer
+{
+    private const int UserStringsArraySize = 100;
+    private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor<HardcodedSecretsAnalyzer>();
+    private static HardcodedSecretsAnalyzer? _instance = null;
+
+    private static bool _started = false;
+
+    private static List<SecretRegex>? _secretRules = null;
+
+    private static ManualResetEventSlim _waitEvent = new ManualResetEventSlim(false);
+
+    public HardcodedSecretsAnalyzer()
+    {
+        Log.Debug("HardcodedSecretsAnalyzer -> Init");
+        LifetimeManager.Instance.AddShutdownTask(RunShutdown);
+        _started = true;
+        Task.Run(() => PoolingThread());
```

Comment:
nit: is it worth adding a continuation here?
Also, we _typically_ store the `Task` and return it as part of a `DisposeAsync` call. You may not want to do that here, I'm more wondering about the shutdown sequence and the implications if you try to submit evidence _after_ the app is already shutting down 🤔 

```suggestion
        Task.Run(() => PoolingThread())
            .ContinueWith(t => Log.Error(t.Exception, "Error in Hardcoded secret analyzer"), TaskContinuationOptions.OnlyOnFaulted);

```

Reply:
I buy that ContinueWith. Fixing it